tests/queries/0_stateless/02661_parameterized_rename_queries.sql

```diff
@@ -0,0 +1,52 @@
+-- Tags: no-parallel
+
+-- Case 1: RENAME DATABASE
+
+DROP DATABASE IF EXISTS 02661_db;
+DROP DATABASE IF EXISTS 02661_db1;
+
+SET param_old_db_name = 02661_db;
+SET param_new_db_name = 02661_db1;
+
+CREATE DATABASE {old_db_name:Identifier};
+RENAME DATABASE {old_db_name:Identifier} TO {new_db_name:Identifier};
+
+SELECT name FROM system.databases WHERE name = {new_db_name:String};
+
+-- Case 2: RENAME TABLE
+
+DROP TABLE IF EXISTS 02661_t;
+DROP TABLE IF EXISTS 02661_t1;
+
+SET param_old_tbl_name = 02661_t;
+SET param_new_tbl_name = 02661_t1;
+
+CREATE TABLE {new_db_name:Identifier}.{old_tbl_name:Identifier} (a UInt64) ENGINE = MergeTree ORDER BY tuple();
```

Comment:
Maybe we should also support something like 
```
SET param_table_name = qualified.name;

CREATE TABLE {table_name:Identifier} (a UInt64) ...
```
(it should create a table named `name` in a database named `qualified`)

Because `qualified.name` is a valid compound identifier.

Reply:
I'm not sure about how it will behave in the case when the table name contains a dot, for example 'table.first' in current implementation will become default.'table.first', and first.'table' if we change it.

Maybe we should implement special parameter type CompoundIdentifier for this?

Reply:
Yes, makes sense

Reply:
Another option is to use `ParserCompoundIdentifier` to parse a value passed to an Identifier parameter but seems like it may be backward incompatible

Reply:
Ok, will do it in a separate PR

Reply:
> Another option is to use ParserCompoundIdentifier to parse a value passed to an Identifier parameter but seems like it may be backward incompatible

Yeah, and it seems like it will only create confusion on the user's part, so a separate type for it looks like the best solution